libbeat/docs/queueconfig.asciidoc

```diff
@@ -76,3 +77,147 @@ will be immediately available for consumption.
 
 The default values is 0s.
 
+[float]
+[[configuration-internal-queue-spool]]
+=== Configure the file spool queue
+
+The file spooling queue stores all events in an on disk ring buffer. The spool
+has a write buffer, which new events are written to. Events written to the
+spool are forwarded to the outputs, only after the write buffer has been
+flushed successfully.
+
+The spool waits for the output to acknowledge or drop events. If the spool is
+full, no new events can be inserted. The spool will block. Space is freed only
+after a signal from the output has been received.
+
+On disk, the spool operates in pages. The `file.page_size` setting configures
+the files page size on creation time. The optimal page size depends on the
+effective block size, used by the underlying file system.
+
+This sample configuration enables the spool with all default settings and the
+default file path:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.spool: ~
+------------------------------------------------------------------------------
+
+This sample configuration creates a spool of 512MiB, with 16KiB pages. The
+write buffer is flushed if 10MiB of contents, or 1024 events have been
+written. If oldest available event is already waiting for 5s in the write
+buffer, the buffer will be flushed as well:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.spool:
+  file:
+    path: "${path.data}/spool.dat"
+    size: 512MiB
+    page_size: 16KiB
+  write:
+    buffer_size: 10MiB
+    flush.timeout: 5s
+    flush.events: 1024
+------------------------------------------------------------------------------
+
+[float]
+==== Configuration options
+
+You can specify the following options in the `queue.spool` section of the
++{beatname_lc}.yml+ config file:
+
+[float]
+===== `file.path`
+
+The spool file path. The file is created on startup, if it does not exist.
+
+The default value is "${path.data}/spool.dat".
+
+===== `file.permissions`
+
+The file permissions. The permissions are applied when the file is
+created. In case the file already exists, The file permissions are compared
+with `file.permissions`. The spool file is not opened if the actual file
```

Comment:
What happens when the spool file cannot be opened? What does the user do? Change the file.permissions to match the actual file, or do they simply specify a different file path? Hopeful the message they see is descriptive enough so they know what to change.

Reply:
The beat won't startup on permissions errors.

File permission error occurs if the on-disk file permissions do not match the required permissions. For example the configuration mandates 'current user only' (permissions flag 0600), but the configuration file is readable/writeable by every user in the sytem (0666).

The error message generated by the spool: 

```
<some message by the beat>: file permissions must be more strict (required permissions: 0600, actual permissions: 0666)
```